packages/core/src/request-builder/header-builder/authorization-header.ts

```diff
@@ -36,178 +23,34 @@ export async function addAuthorizationHeader<
-      ...map(token => token.error, authTokens)
-    ].join('\n')
-  );
-};
-
-const headerForOnPremProxyAuth = (destination: Destination): MapType<any> => {
-  if (destination.proxyType !== 'OnPremise') {
-    return {};
-  }
-  // The connectivity service will raise an exception if it can not obtain the 'Proxy-Authorization' and the destination lookup will fail early
-  return {
-    'Proxy-Authorization': destination!.proxyConfiguration!.headers![
-      'Proxy-Authorization'
-    ]
-  };
-};
-
-const headerForUserPropagation = (destination: Destination): MapType<any> => {
-  const proxyHeaders = destination?.proxyConfiguration?.headers;
-  if (!proxyHeaders || !proxyHeaders['SAP-Connectivity-Authentication']) {
-    throw new Error(
-      'Principal propagation was selected in destination, but no SAP-Connectivity-Authentication bearer header was added by connectivity-service.'
-    );
-  }
-  return {
-    'SAP-Connectivity-Authentication':
-      proxyHeaders['SAP-Connectivity-Authentication']
+export function buildAndAddAuthorizationHeader(destination: Destination) {
+  return async function (headers: MapType<any>): Promise<MapType<string>> {
+    return {
+      ...headers,
+      ...(await buildAuthorizationHeader(destination))
+    };
   };
-};
-
-const headerFromOAuth2ClientCredentialsDestination = (
-  destination: Destination
-): Promise<string> =>
-  getOAuth2ClientCredentialsToken(destination)
-    .then(resp => `Bearer ${resp.access_token}`)
-    .catch(error =>
-      Promise.reject(
-        errorWithCause(
-          'Request for "OAuth2ClientCredentials" authentication access token failed or denied.',
-          error
-        )
-      )
-    );
-
-const headerFromAuthToken = (token: DestinationAuthToken) =>
-  `${token.type} ${token.value}`;
-
-// Using pipe led to wrong type errors
-const headerFromTokens = (authTokens: DestinationAuthToken[]): string => {
-  const usableTokens = filter(
-    (token: DestinationAuthToken) => !token.error,
-    authTokens
-  );
-
-  if (authTokens === null) {
-    throw new Error(
-      'AuthenticationType is "OAuth2SAMLBearerAssertion", but no AuthTokens could be fetched from the destination service!'
-    );
-  }
-  if (isEmpty(usableTokens)) {
-    throwAllTokensErrored(authTokens);
-  }
-
-  ifElse(isEmpty, throwAllTokensErrored, identity)(usableTokens);
-
-  const usableToken = head(usableTokens);
-  if (!usableToken) {
-    throw new Error(`No usable tokens are found in the ${usableTokens}`);
-  }
-  return headerFromAuthToken(usableToken!);
-};
-
-const headerFromBasicAuthDestination = (destination: Destination) => {
-  if (isNil(destination.username) || isNil(destination.password)) {
-    throw new Error(
-      'AuthenticationType is "BasicAuthentication", but "username" and/or "password" are missing!'
-    );
-  }
-
-  return basicHeader(destination.username, destination.password);
-};
-
-export function basicHeader(username: string, password: string): string {
```

Comment:
Question: When this method is moved to the auth-header.ts file. Isn't it a breaking change?

Reply:
If so also keep it and depricate it and move a copy to auth-header?

Reply:
Unfortunately, yes. But I cannot export the same function twice (because it is exported from the index.ts). Not sure what the best option would be here... I could also keep in the old file and rename the other, but I don't love this.

Reply:
I would keep it in the old file. You never know who uses it and gets annoyed be a change.

Reply:
I put everything back in the old files for auth and csrf.